src/cgr_gwas_qc/workflow/scripts/concordance_table.py

```diff
@@ -58,16 +69,65 @@ def read(filename: PathLike) -> pd.DataFrame:
     return pd.read_csv(filename, dtype=DTYPES)
 
 
-def build(filename: Path, concordance_threshold: float, pi_hat_threshold: float) -> pd.DataFrame:
+def _sort_ids(x: pd.DataFrame):
+    """ "Sort IDs alphanumerically."""
+    x.IID1, x.IID2 = np.where(x.IID1 < x.IID2, [x.IID1, x.IID2], [x.IID2, x.IID1])
+    x.rename(columns={"IID1": "ID1", "IID2": "ID2"}, inplace=True)
+    return x
+
+
+def prep_concordance_table(pi_hat_threshold: float, concordance_threshold: float, x: pd.DataFrame):
+    """ "Prepares the concordance_table by appending new columns based on thresholds."""
     return (
-        plink.read_genome(filename)
+        _sort_ids(x)
         .assign(is_ge_pi_hat=lambda x: x.PI_HAT >= pi_hat_threshold)
         .assign(concordance=lambda x: x.IBS2 / (x.IBS0 + x.IBS1 + x.IBS2))
         .assign(is_ge_concordance=lambda x: x.concordance >= concordance_threshold)
         .reindex(DTYPES.keys(), axis=1)
     )
 
 
+def process_genome_chunk(pi_hat_threshold: float, concordance_threshold: float, chunk):
+    """ "Processes each chunk of genome file."""
+    temp_file = tempfile.NamedTemporaryFile(delete=False)
+    prep_concordance_table(pi_hat_threshold, concordance_threshold, chunk).to_csv(
+        temp_file.name, index=False, header=False
+    )
+    return temp_file.name
+
+
+def build(
+    filename: Path,
+    concordance_threshold: float,
+    pi_hat_threshold: float,
+    outfile: Path,
+    threads: int,
+):
+
+    plink_genome_file = plink.read_genome(
+        filename, required_cols=["IID1", "IID2", "PI_HAT", "IBS0", "IBS1", "IBS2"], chunksize=100000
+    )
+
+    temp_files = []
+
+    header = tempfile.NamedTemporaryFile(delete=False)
+    temp_files.append(header.name)
+    pd.DataFrame(columns=DTYPES.keys()).to_csv(header.name, index=False)
+
+    with concurrent.futures.ProcessPoolExecutor(max_workers=threads) as executor:
+        results = executor.map(
+            functools.partial(process_genome_chunk, pi_hat_threshold, concordance_threshold),
+            plink_genome_file,
+        )
+
+    temp_files = temp_files + list(results)
+    with open(outfile, "wb") as outputfile:
+        subprocess.run(["cat"] + temp_files, stdout=outputfile)
+
+    for f in temp_files:
+        Path(f).unlink
```

Comment:
Don't forget parenthesis to this method, else it won't remove the temp file.
Should be `Path(f).unlink()`

Reply:
Good catch. Fixed in 7a146d76a8a0fe6e228e5d1a68011be080a297c6